internal/registration/internal/application/command_handler.go

```diff
@@ -0,0 +1,94 @@
+package application
+
+import (
+	"GoEdu/internal/registration/internal/domain"
+	"context"
+	"fmt"
+)
+
+type CommandHandler interface {
+	Handle(context.Context, Command) (CommandResult, error)
+}
+
+type RegisterNewUserCommandHandler struct {
+	hasher     PasswordHasher
+	verifier   UniqueEmailVerifier
+	repository UserRegistrationRepository
+}
+
+func (r *RegisterNewUserCommandHandler) Handle(context context.Context, command Command) (CommandResult, error) {
+	regNewUserCommand := command.(RegisterNewUserCommand)
+
+	h, err := r.hasher.Hash(regNewUserCommand.Password)
+	if err != nil {
+		return nil, err
+	}
+
+	user, err := domain.RegisterNewUser(
+		regNewUserCommand.FirstName,
+		regNewUserCommand.LastName,
+		regNewUserCommand.Email,
+		h,
+	)
+	if err != nil {
+		return nil, err
+	}
+
+	if res, err := r.verifier.IsUnique(context, user.Email); err != nil || res != true {
```

Comment:
Also I think it's valid to call this check before you call domain logic to register new user. This will force you to construct Email VO somewhere else. For example at the interface layer. So you can make the command consist already prepared and validated email before it will go to the application layer.